### PR TITLE
Add curriculum_umbrella property to scripts 

### PR DESCRIPTION
**Pull request by Erin007**
Work for: [LP-386](https://codedotorg.atlassian.net/browse/LP-386)

This PR adds the ability for Levelbuilders to add a new property, `curriculum_umbrella` to a script that currently can be set to nil, 'CSF', 'CSD' or 'CSP'.  This property will allow us to more easily reference any script that is part of one of these curricula regardless of version year, which means less code to maintain and update, such as [hardcoded lists of CSF scripts](https://github.com/code-dot-org/code-dot-org/blob/3cea1f1ff168e5b0ef294befe337819acf69f58c/lib/cdo/script_config.rb#L28),  when we switch versions. 

curriculum_umbrella_dropdown

Naming things is tricky, especially as it relates to scripts/courses, in part because we have different names across the org for the same concept. Here's a [handy summary of current course terminology](https://docs.google.com/spreadsheets/d/1naiSH_RfWw-RijhbWKy4ZwNOctZBtd2Cm7RegL0bPdY/edit#gid=0). 

 I selected `curriculum_umbrella` because I think: 
1.) it avoids continued overloading/confusion with the word "course" 
2.) umbrella immediately denotes that this is a large, catchall category 
3.) it directly references curriculum, which is how the curriculum team refers to CSF/CSD/CSP and this new term will be used in the context of Levelbuilder, which is their tool 
4.) it's big enough and fresh enough that in could, in the future, hold CSX, or some other not-yet-invented "chunk" of content.

### CODE REVIEW

File: dashboard/test/dsl/dsl_test.rb [hunk 1]
```diff
@@ -628,38 +629,31 @@ def setup
     assert_equal expected, script_text
   end
 
-  test 'Script DSL with new_name, family_name, version_year and is_stable' do
```
Comment (davidsbailey) [inline dashboard/test/dsl/dsl_test.rb]: restore accidentally removed test?

**Comment by Erin007** [comment]
I think this property has the potential to let us clean up a few other places in addition to the hardcoded list in `script_config`.  For example, I think we can use it in lieu of the `excludeCsfColumnInLegend` field and in the logic where we determine if the [congrats page should be shown](https://github.com/code-dot-org/code-dot-org/blob/3cea1f1ff168e5b0ef294befe337819acf69f58c/lib/cdo/script_constants.rb#L276). Still digging, but if you have other areas for potential clean up or cautionary feedback for the areas I've already identified, let me know please!

File: apps/src/lib/script-editor/ScriptEditor.jsx [hunk 2]
```diff
@@ -140,6 +143,23 @@ export default class ScriptEditor extends React.Component {
...
+        <label>
+          Is this script part of one of the core courses?
+          <select
+            name="curriculum_umbrella"
+            style={{marginLeft: 20}}
+            defaultValue={this.props.curriculumUmbrella}
+            ref={select => (this.curriculumUmbrellaSelect = select)}
+          >
+            {CURRICULUM_UMBRELLAS.map(curriculumUmbrella => (
+              <option value={curriculumUmbrella}>{curriculumUmbrella}</option>
```
Comment (davidsbailey) [inline apps/src/lib/script-editor/ScriptEditor.jsx]: I think these `option` tags need a `key` attribute to make React happy

File: dashboard/app/models/script.rb [hunk 3]
```diff
@@ -1623,4 +1626,16 @@ def self.has_any_pilot_access?(user = nil)
...
+  def csf?
+    return true if curriculum_umbrella == 'CSF'
+  end
+
+  def csd?
+    return true if curriculum_umbrella == 'CSD'
+  end
+
+  def csp?
+    return true if curriculum_umbrella == 'CSP'
```
Comment (davidsbailey) [inline dashboard/app/models/script.rb]: these helper methods are a great touch!

**Review by davidsbailey — APPROVED** [review]
👏 👏 👏 fantastic work, @Erin007 ! Really nice design choices throughout!

**Comment by davidsbailey** [comment]
> I think this property has the potential to let us clean up a few other places in addition to the hardcoded list in `script_config`. For example, I think we can use it in lieu of the `excludeCsfColumnInLegend` field and in the logic where we determine if the [congrats page should be shown](https://github.com/code-dot-org/code-dot-org/blob/3cea1f1ff168e5b0ef294befe337819acf69f58c/lib/cdo/script_constants.rb#L276). Still digging, but if you have other areas for potential clean up or cautionary feedback for the areas I've already identified, let me know please!

Agreed this has great potential for cleanup, and that there could be pitfalls :-) The one that comes to mind is that each codepath that checks whether something is e.g. a CSF script may only want those which are un-hidden, stable, recommended, etc. This is still totally doable -- it just means you'll probably need to pass `hidden` and `is_stable` to some of the places you pass `curriculum_umbrella`, and also check with feature owners what threshold they want for when to include the newest scripts.

File: dashboard/test/controllers/scripts_controller_test.rb [hunk 4]
```diff
@@ -276,6 +278,7 @@ class ScriptsControllerTest < ActionController::TestCase
     assert script.login_required
     assert script.hideable_stages
     assert script.project_sharing
+    asser script.curriculum_umbrella
```
Comment (maddiedierker) [inline dashboard/test/controllers/scripts_controller_test.rb]:
```suggestion
    assert_equal 'CSP', script.curriculum_umbrella
```

**Review by maddiedierker — APPROVED** [review]
really great job! i think there will be a lot of places where this change eliminates complexity 🥇

File: dashboard/app/models/script.rb (hunk 3 above)
Comment (Erin007) [reply dashboard/app/models/script.rb]: Turns out we already have a helper `csf?` https://github.com/code-dot-org/code-dot-org/blob/529afde8ab4379b9fb2865efe5d27ac50badd3c3/dashboard/app/models/script.rb#L689, which means that `k5_course?` is yet another place where we have a hardcoded list of of csf versions and another candidate for cleaning up with the new property. I'm going to remove the helpers for now, and add them in a later PR where they will be used and not overwrite the current method.

**Pull request merged**